src/jupyter_contrib_nbextensions/nbconvert_support/pre_embedimages.py

```diff
@@ -0,0 +1,157 @@
+    Valid resize settings are: small = 500px, mid = 1000px, large = 2000px
+    for maximum size in length or width.  No upscaling of small images will
+    be performed. The Python package `PIL` needs to be installed for this
+    option to work.
+
+    Example::
+
+            $ jupyter nbconvert --to html --EmbedImagesPreprocessor.embed_images=True
+                --EmbedImagesPreprocessor.resize=large mynotebook.ipynb
+
+    *Note:* To embed images after conversion to HTML you can also use the
+           `html_embed` exporter
+    """
+
+    embed_images = Bool(False, help="Embed images as attachment").tag(config=True)
+    embed_remote_images = Bool(False, help="Embed images referenced by an url as attachment").tag(config=True)
+    resize = Unicode('', help="Resize images to save space (reduce size)").tag(config=True)
+    imgsizes = {'small': 500, 'mid': 1000, 'large': 2000}
+
+    def preprocess(self, nb, resources):
+        """Skip preprocessor if not enabled"""
+        if self.embed_images:
+            nb, resources = super(EmbedImagesPreprocessor, self).preprocess(nb, resources)
+        return nb, resources
+
+    def resize_image(self, imgname, imgformat, imgdata):
+        """Resize images if desired and PIL is installed
+
+        Parameters
+        ----------
+            imgname: str
+                Name of image
+            imgformat: str
+                Format of image (JPG or PNG)
+            imgdata:
+                Binary image data
+
+        """
+        if imgformat in ['png', 'jpg']:
+            from io import BytesIO
+            try:
+                from PIL import Image
+            except ImportError:
+                self.log.info("Pillow library not available to resize images")
+                return imgdata
+            # Only make images smaller when rescaling
+            im = Image.open(BytesIO(imgdata))
+            factor = self.imgsizes[self.resize] / max(im.size)
+            if factor < 1.0:
+                newsize = (int(im.size[0] * factor), int(im.size[1] * factor))
+                newim = im.resize(newsize)
+                fp = BytesIO()
+                # PIL requires JPEG instead of JPG
+                newim.save(fp, format=imgformat.replace('jpg', 'jpeg'))
+                imgdata = fp.getvalue()
+                fp.close()
+                self.log.debug("Resized %d x %d image %s to size %d x %d pixels" %
+                               (im.size[0], im.size[1], imgname, newsize[0], newsize[1]))
+        return imgdata
+
+    def replfunc_md(self, match):
+        """Read image and store as base64 encoded attachment"""
+        url = match.group(2)
+        imgformat = url.split('.')[-1].lower()
+        if url.startswith('http'):
+            if self.embed_remote_images:
+                data = urlopen(url).read()
+            else:
+                return match.string
+        elif url.startswith('attachment'):
+            return match.string
```

Comment:
`[asd](attachement:asd.jpg) # this comment...`
will result here in 
`[asd](attachement:asd.jpg) # this comment... # this comment...`
it should be `match.group(0)`

Reply:
match.string is not what should be returned if we do not want to substitute...
match.string is the string passed to match/search... https://docs.python.org/3/library/re.html#re.match.string